eng/common/scripts/job-matrix/pullrequest-faq.md

```diff
@@ -0,0 +1,89 @@
+# Pull Request FAQ
+
+The `pullrequest` pipeline is a single public definition that handles all `pull request` changes to an `azure-sdk-for-X` repository. This document is intended to answer some common questions that users may have about the `pullrequest` definition.
+
+## Can I get a bit more context first?
+
+Let's get some basic repo structure discussion out of the way. The `azure-sdk` team maintains a consistent repo structure for all shipping packages to package managers (Read NPM, Nuget, pypi, Maven, etc)
+
+```jsonc
+sdk/
+  storage
+    Azure.Storage.Blobs
+    Azure.Storage.Queues
+    ...
+  <service>
+    <service-package-1>
+    ..
+    <service-package-N>
+    // the ci.yml is what AZDO build defs are based upon
+    ci.yml
+```
+
+This necessitates that release definitions on the [internal](https://dev.azure.com/azure-sdk/internal/) azure devops exist for each service in a repository. However, each build definition can only build and ship packages **within the service it was created for**.
+
+This service-directory also applied to  `public` build definitions that triggered on [pull requests](https://github.com/Azure/azure-sdk-for-python/pulls) in our repos. Due to this, large changesets that touched multiple service directories would incurr a build for _every servicedirectory that was touched_. The `azure-sdk` EngSys calls this situation a `build storm`.
+
+The `<language> - pullrequest` definitions entirely replace service-specific build definitions. It has the ability to expand and contract the targeted packagesfor build according to a **git diff** of the actual changes made. Because of this, any repo that has cut over to `pullrequest` will enjoy no longer incurring build storms on large cross-cutting changes. While the individual build run will be very long running and batch up tests across a bunch of agents, it _will_ eventually complete. It will be _impossible_ to exhaust Github or Devops token utilization as well, given that it is a single definition triggering checks.
```

Comment:
```suggestion
The `<language> - pullrequest` definitions entirely replace service-specific build definitions. It has the ability to expand and contract the targeted packages for build according to a **git diff** of the actual changes made. Because of this, any repo that has cut over to `pullrequest` will enjoy no longer incurring build storms on large cross-cutting changes. While the individual build run will be very long running and batch up tests across a bunch of agents, it _will_ eventually complete. It will be _impossible_ to exhaust GitHub or Azure Dev Ops token utilization as well, given that it is a single definition triggering checks.
```